### ISSUE TITLE
Healthcheck / Readiness probe

### ISSUE THREAD

**Issue by LukasKnuth** [outofcoffee/imposter#460]
I'm using imposter in my end to end test setup via Docker Compose. Since the application to test depends on making requests to the mocked service quite early, I'd like to start it only when the mock server is up and ready.

In Docker Compose files, you can do this by adding a `healthcheck` condition to the service and then depend on the `service_healthy` status in the `depends_on` declaration:

```
healthcheck:
  test: ["CMD-SHELL", "curl -f http://imposter:8080 || exit 1"]
  interval: 5s
  timeout: 5s
  retries: 10
```

I have two questions:

1. The command given in `test` is executed _inside_ the container, so any tool to do the actual health check must be in the container. Since the container is quite locked  down, I haven't found anything I could use for this purpose. Would you be open to adding a CLI option to the server to check if its up and ready?
2. If not, is there a good route to query for readiness? I haven't seen the classic `/health` endpoint.

**Comment by outofcoffee** [comment]
Hi @LukasKnuth, thanks for raising this.

Yes, you can use the `/system/status` [healthcheck endpoint](https://docs.imposter.sh/metrics_logs_telemetry/) which returns `HTTP 200` when the server is up.

For additional binaries inside the container, here's an example of a custom Docker image https://github.com/outofcoffee/imposter/blob/main/examples/docker/Dockerfile

☝️ This example extends from the Java base image so has its various userspace tools available to install.

**Comment by outofcoffee** [comment]
For this part of your question:

> Would you be open to adding a CLI option to the server to check if its up and ready?

You should be able to use this command with the CLI:

```
imposter list --exit-code-health
```

This sets the exit code based on mock health.

[More info here](https://github.com/gatehill/imposter-cli#list-all-running-mocks).

**Comment by LukasKnuth** [comment]
Nice, thanks for getting back to me. I wasn't aware that this existed already, thats great!

**Comment by LukasKnuth** [comment]
I tested it with my setup now and for me, this doesn't seem to work as expected:

```bash
$ docker exec e2e-apiv1-1 imposter list -x
| ID | NAME | PORT | HEALTH  |
|----|------|------|---------|
|  1 | java |    0 | unknown |
$ echo $?
1
```

I use the `openapi` plugin for this. Everything in the mock works as expected, just the list returns the `unknown` health status. Do I need to add any additional information to the configuration for `imposter ls` to pick it up and/or it's health?

**Comment by outofcoffee** [comment]
Hi @LukasKnuth, thank you for reporting this.

Sorry to hear you had an issue with the healthcheck. This was fixed in [v0.34.0 of the CLI](https://github.com/gatehill/imposter-cli/releases), which was released in [Imposter 3.31.6](https://github.com/outofcoffee/imposter/blob/main/CHANGELOG.md#3316---2023-10-26), released on 26/10/23.

**Comment by rnestler** [comment]
If somebody ends here via search: One can configure a healthcheck with docker-compose like this:
```yaml
    healthcheck:
      test: ["CMD", "imposter", "list", "-x"]
      interval: 3s
      timeout: 10s
      retries: 5
```
